tutorial/08-diffusion.ipynb

```diff
@@ -147,79 +220,127 @@
     "    ny = len(m.structured_get_divisions(\"y\"))\n",
     "    nz = len(m.structured_get_divisions(\"z\"))\n",
     "    shape = (nx, ny, nz)\n",
-    "    D = m.mesh.getTagHandle(\"D\")\n",
-    "    k = m.mesh.getTagHandle(\"k\")\n",
-    "    S = m.mesh.getTagHandle(\"S\")\n",
-    "    Sigma_a = m.mesh.getTagHandle(\"Sigma_a\")\n",
-    "    phi = m.mesh.getTagHandle(\"flux\")\n",
-    "    phi_next = m.mesh.getTagHandle(\"phi_next\")\n",
-    "    for idx in product(*[range(xyz-1) for xyz in shape]):\n",
+    "    D = m.mesh.tag_get_handle(\"D\")\n",
+    "    k = m.mesh.tag_get_handle(\"k\")\n",
+    "    S = m.mesh.tag_get_handle(\"S\")\n",
+    "    Sigma_a = m.mesh.tag_get_handle(\"Sigma_a\")\n",
+    "    phi = m.mesh.tag_get_handle(\"flux\")\n",
+    "    phi_next = m.mesh.tag_get_handle(\"phi_next\")\n",
+    "    for idx in product(*[range(xyz - 1) for xyz in shape]):\n",
     "        ent = m.structured_get_hex(*idx)\n",
-    "        phi_next[ent] = (max(D[ent] * laplace(phi, idx, m, shape) + \n",
-    "                                    (k[ent] - 1.0) * Sigma_a[ent] * phi[ent], 0.0) + S[ent])*dt*2.2e5 + phi[ent]\n",
-    "    ents = m.mesh.getEntities(iBase.Type.region)\n",
-    "    phi[ents] = phi_next[ents]"
+    "        # Get the value of phi for the entity\n",
+    "        phi_val = m.mesh.tag_get_data(phi, ent)[0]\n",
+    "        # Get the value of D for the entity\n",
+    "        D_val = m.mesh.tag_get_data(D, ent)[0]\n",
+    "        # Get the value of k for the entity\n",
+    "        k_val = m.mesh.tag_get_data(k, ent)[0]\n",
+    "        # Get the value of Sigma_a for the entity\n",
+    "        Sigma_a_val = m.mesh.tag_get_data(Sigma_a, ent)[0]\n",
+    "        # Get the value of S for the entity\n",
+    "        S_val = m.mesh.tag_get_data(S, ent)[0]\n",
+    "\n",
+    "        # Perform calculations using the retrieved values\n",
+    "        phi_next_val = (max(\n",
+    "            D_val * laplace(phi, idx, m, shape) +\n",
+    "            (k_val - 1.0) * Sigma_a_val * phi_val, 0.0\n",
+    "        ) + S_val) * dt * 2.2e5 + phi_val\n",
```

Comment:
You did not introduce this, but it’s a good time for a change.  Can we define a variable `v` or `speed` that is defined as `2.2e5` for clarity? And then add in the comments that this is a one-speed solution for thermal neutrons?

Reply:
I have added this new variable.